includes/class-wc-payments-token-service.php

```diff
@@ -137,12 +140,13 @@ public function woocommerce_get_customer_payment_tokens( $tokens, $user_id, $gat
 
 		// Prevent unnecessary recursion, WC_Payment_Token::save() ends up calling 'woocommerce_get_customer_payment_tokens' in some cases.
 		remove_action( 'woocommerce_get_customer_payment_tokens', [ $this, 'woocommerce_get_customer_payment_tokens' ], 10, 3 );
+
 		foreach ( $payment_methods as $payment_method ) {
 			if ( ! isset( $payment_method['type'] ) ) {
 				continue;
 			}
 
-			if ( ! isset( $stored_tokens[ $payment_method['id'] ] ) ) {
+			if ( ! isset( $stored_tokens[ $payment_method['id'] ] ) && ( 'card' === $payment_method['type'] && WC_Payment_Gateway_WCPay::GATEWAY_ID === $gateway_id || 'sepa_debit' === $payment_method['type'] && WC_Payment_Gateway_WCPay::GATEWAY_ID . '_' . Payment_Method::SEPA === $gateway_id ) || empty( $gateway_id ) ) {
```

Comment:
```suggestion
			if ( ! isset( $stored_tokens[ $payment_method['id'] ] ) && ( Payment_Method::CARD=== $payment_method['type'] && WC_Payment_Gateway_WCPay::GATEWAY_ID === $gateway_id || Payment_Method::SEPA === $payment_method['type'] && WC_Payment_Gateway_WCPay::GATEWAY_ID . '_' . Payment_Method::SEPA === $gateway_id ) ) {
```
Might as well use all the constants we've got. Also, consider breaking this up into a multi-line statement: this was a challenging read. 😅 

Also, do we need that last `empty( $gateway_id )` check? Presumably, that's not possible, if we've made it this far.